tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -374,49 +453,52 @@ func addDeleteStaticRouteOnWorkerNodes(testPodList []*pod.Builder, routeMap map[
 	}
 }
 
-func activateNftablesIfInactive() {
+func activateNftablesIfInactive(targetNodeName string) {
 	// Get the current status of the nftables service
-	statuses, err := cluster.ExecCmdWithStdoutWithRetries(APIClient,
-		3, 10*time.Second,
-		"systemctl is-active nftables.service | cat -")
+	statuses, err := cluster.ExecCmdWithStdout(APIClient,
+		"systemctl is-active nftables.service | cat -",
+		metav1.ListOptions{LabelSelector: "kubernetes.io/hostname=worker-0"})
+	fmt.Println("############################### ENABLED", statuses)
 	Expect(err).ToNot(HaveOccurred(), "Failed to check if nftables service status")
 	Expect(statuses).ToNot(BeEmpty(), "Failed to find statuses for nftables service")
 
 	// Iterate through the statuses of the nodes
 	for nodeName, status := range statuses {
-		// If the status is inactive, activate it
+		// If the node matches targetNodeName and the status is inactive, activate it
 		status = strings.TrimSpace(status)
-		if status == "inactive" {
+		if nodeName == targetNodeName && status == "inactive" {
```

Comment:
Why do we need nodeName validation if you use metav1.ListOptions{LabelSelector: "kubernetes.io/hostname=worker-0"}?

Reply:
correct I dont need. removed